src/structs/ortho.rs

```diff
@@ -232,3 +247,20 @@ impl<N: BaseFloat + Clone> OrthoMat3<N> {
         self.mat.clone()
     }
 }
+
+#[cfg(feature="arbitrary")]
+impl<N: Arbitrary + BaseFloat> Arbitrary for OrthoMat3<N> {
+    fn arbitrary<G: Gen>(g: &mut G) -> OrthoMat3<N> {
+        let x: Ortho3<N> = Arbitrary::arbitrary(g);
+        x.to_persp_mat()
+    }
+}
+
+
+/// Simple helper function for rejection sampling
+#[cfg(feature="arbitrary")]
+#[inline]
+pub fn reject<G: Gen, F: FnMut(&T) -> bool, T: Arbitrary>(g: &mut G, f: F) -> T {
+    use std::iter::repeat;
+    repeat(()).map(|_| Arbitrary::arbitrary(g)).filter(f).next().unwrap()
```

Comment:
Functional programming style… I just hope the compiler strips out all the fluff it generates. Otherwise this could be easily converted to a loop.